src/common/selftest/selftest_firstlayer.hpp

```diff
@@ -0,0 +1,91 @@
+/**
+ * @file selftest_firstlayer.hpp
+ */
+#pragma once
+
+#include <inttypes.h>
+#include "i_selftest_part.hpp"
+#include "selftest_firstlayer_config.hpp"
+#include "selftest_log.hpp"
+#include "../../marlin_stubs/G26.hpp"
+
+namespace selftest {
+
+/**
+ * @brief Object handling states of first layer calibration
+ *
+ * Simplified state-change map of first layer
+ *
+ * AskFilament --> Calib / Preheat / Load / Unload
+ * Calib       --> Finish / Calib
+ * Preheat     --> Calib
+ * Load        --> Calib / AskFilament
+ * Unload      --> AskFilament
+ */
+class CSelftestPart_FirstLayer {
+    enum {
+        FKNOWN = 0x01,      //filament is known
+        F_NOTSENSED = 0x02, //filament is not in sensor
```

Comment:
Scope of those enums is too big. It is used only in `CSelftestPart_FirstLayer::stateAskFilamentInit()` so maybe it would be beneficial to define it inside of that method. Maybe the code would become more readable.

Reply:
rewritten